src/doc/reference.md

```diff
@@ -3769,7 +3769,20 @@ impl Printable for String {
 
 The notation `&self` is a shorthand for `self: &Self`. In this case,
 in the impl, `Self` refers to the value of type `String` that is the
-receiver for a call to the method `make_string`.
+receiver for a call to the method `make_string`. Thus, it could also
+have been written:
+
+```
+# trait Printable {
+#     fn make_string(&self) -> String;
+# }
+# 
+impl Printable for String {
+    fn make_string(&self) -> Self {
```

Comment:
Yeah, this may not be best example.

The purpose of the PR is that when I first ran in to `Self` in the wild (not `self`) I was confused how it worked since I hadn't seen something like that in another language.  When I came to the docs, the examples used `self` but not `Self` directly.  I mostly just wanted to get some example into the reference that actually used `Self` directly so that someone else new would looked it up would immediately see it used in an example.  It is explained well in the paragraph, but not using it directly in an example is what I was trying to fill.

Is there a better way to show an example of actually using `Self` directly, or is this an anti-pattern all together?